File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/input/PerceiveActionImpulse.kt
package gaia.sdk.request.input

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

/**
 * Input for action perception impulse
 */
class PerceiveActionImpulse(val broadcast:Boolean, 
   val identityId:Uuid, 
   val type:String, 
   val data:Struct) : Input()

File: gaia-sdk-javascript/src/graphql/request/input/PerceiveActionImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Input for action perception impulse
 */
export class PerceiveActionImpulse {
public _typeName = "PerceiveActionImpulse";
    private broadcast:Boolean;
    private identityId:Uuid;
    private type:String;
    private data:Struct;

    constructor (broadcast:Boolean, identityId:Uuid, type:String, data:Struct) {
        this.broadcast = broadcast;
        this.identityId = identityId;
        this.type = type;
        this.data = data;
    }
}
